### ISSUE TITLE
OSX - Migration from Version 7.1.0 to 8 - json-fortran reads integers greater than 9 into 1

### ISSUE THREAD

**Issue by gh4ag** [jacobwilliams/json-fortran#459]
## Full Report 

See issue reported [here on hombrew for full detail](https://github.com/Homebrew/homebrew-core/issues/55280) with complete MWE.

## Summary:

The error is that any integer greater than 9 will be parsed as 1 when loading a json file when using the version 8 of `json-fortran`.

Basically, Homebrew only allows users to install son-fortran version 8 compiled with GCC 9.3 using the following config (https://github.com/Homebrew/homebrew-core/blob/master/Formula/json-fortran.rb) :

> def install
>     mkdir "build" do
>       system "cmake", "..", *std_cmake_args,
>                             "-DUSE_GNU_INSTALL_CONVENTION:BOOL=TRUE",
>                             "-DENABLE_UNICODE:BOOL=TRUE"
>       system "make", "install"
>     end
>   end

Installing the V8 from source resulted to the same behavior. 

However, reverting to V7.1 even with GCC 9.3 fixed the issue.

**Comment by jacobwilliams** [comment]
Very odd. 

I don't see this behavior using my local gfortran 9.1.0. And the TravisCI is testing 9.2.1 and doesn't seem to have any problems. Maybe it a gfortran compiler bug of some kind with 9.3?

**Comment by jacobwilliams** [comment]
What happens if you run this version of the main program?:
```fortran
PROGRAM Test
    use json_module, ik => json_ik, rk => json_rk, lk => json_lk
    type(json_file) :: json
    logical(kind=lk) :: found
    real(kind=rk), dimension(:), allocatable :: rArray
    integer(kind=ik), dimension(:), allocatable :: val    !! value

    print *, "Reading json..."

    ! initialize the class
    call json%initialize(strict_integer_type_checking=.true.)
    if (json%failed()) error stop 'init error'

    ! read the file
    call json%load(filename='file.json')
    if (json%failed()) error stop 'load error'

    ! retreive value from json
    call json%get('data.num', val, found)
    if (json%failed()) error stop 'get error'

    print *, "Num: ", int(val)

    ! print the file to the console
    call json%print()

    ! clean up
    call json%destroy()
    if (json%failed()) stop 1

END PROGRAM Test
```

**Comment by jacobwilliams** [comment]
I can reproduce the issue with 9.3 now, with the binaries provided by home-brew (but not the ones I compile myself for some reason)

It seems if I remove the `-std=legacy` it works. Can you confirm?

**Comment by gh4ag** [comment]
You nailed it! Removing the flag "-std=legacy" solved the issue as well with the version 8.0 and GCC 9.3.

I had reused a colleague's code that used that flag and I used it to reproduce the issue. As you have mentioned it, it shouldn't be useful anymore and I will update out code accordingly.

If you agree, I would consider this issue closed then. 

One thing though, on Homebrew, it is possible for some formulae to access previous version using `brew install formula@X.Y` where X.Y is the version number. Would it make sense for json-fortran to offer previous versions in the same manner or you would consider that the installation from source is enough?

Thank you for your swift response!
